core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -309,6 +328,48 @@ private void releaseBlockLockIfUnused(long blockId) {
   }
 
   /**
+   * Checks the internal state of the manager to make sure invariants hold.
+   *
+   * This method is intended for testing purposes. A runtime exception will be thrown if invalid
+   * state is encountered.
+   */
+  public void validate() {
```

Comment:
I am concerned about having such testing-only logic in production code. Can we move it to test code and use reflection to access the private members?